lib/datadog/tracing/correlation.rb

```diff
@@ -23,7 +23,6 @@ class Identifier
           :span_resource,
           :span_service,
           :span_type,
-          :trace_id,
```

Comment:
I think removing `trace_id` from the public api of this class is problematic.

I think a better solution is to allow `trace_id` to continue to be called, but return a string when `Datadog.configuration.tracing.trace_id_128_bit_logging_enabled` is true. This means users that opt into this new feature get different behavior, but only after enabling this option setting.